### PR TITLE
B/command

### PR DESCRIPTION
**Pull request by worldofjoni**
# Pull Request for initial Implementation

Changes to command and surroundings

## Backend
Make sure, you have done de following before merging this pull request:
- [x] Your code is **documented** using doc comments.
- [x] You added appropriate **logging** in your code.
- [x] You added **unit tests** to all your implemented functions.
- [x] You ran `cargo format` to format your code.
- [x] You ran `cargo clippy` to check for improvements.

### CODE REVIEW

File: backend/src/interface/persistent_data.rs [hunk 1]
```diff
@@ -185,22 +167,26 @@ pub trait CommandDataAccess: Sync + Send {
...
-        meal_id: Uuid,
-        user_id: Uuid,
-        image_hoster_id: String,
-        url: String,
-    ) -> Result<()>;
+    async fn link_image(&self, meal_id: Uuid, user_id: Uuid) -> Result<Uuid>;
+
+    /// Reverts the linking of the given image by deleting the link.
+    /// Useful if an error ocurred with the image itself.
+    async fn revert_link_image(&self, image_id: Uuid) -> Result<()>;
```
Comment (Whatsuup) [inline backend/src/interface/persistent_data.rs:174]: Is there a usage now? For my understanding, images will be linked after evaluation (image_validation component). Users are not able to remove images by themselves. If it is unused we should mark it as unused (for now)

File: backend/src/layer/data/database/auth.rs [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+use sqlx::{Pool, Postgres};
+
+use crate::interface::persistent_data::{model::ApiKey, AuthDataAccess, Result};
+
+/// Class implementing all database requests arising from graphql manipulations.
+#[derive(Debug)]
+pub struct PersistentAuthData {
+    pub(super) pool: Pool<Postgres>,
+}
+
```
Comment (Whatsuup) [inline backend/src/layer/data/database/auth.rs:13]: Do we want a timestamp column for the api_keys? Or is their lifetime not that important? Idk the auth process that well, if this makes sense...

File: backend/src/layer/logic/api_command/command_handler.rs [hunk 3]
```diff
@@ -140,71 +132,57 @@ where
...
+
+        // verify with api
+        self.image_validation.validate_image(&image).await?;
+
+        // link in database
+        let image_id = self.command_data.link_image(meal_id, client_id).await?;
+
+        // store to disk
+        if let Err(e) = self.image_storage.save_image(image_id, image).await {
+            self.command_data.revert_link_image(image_id).await?;
```
Comment (Whatsuup) [inline backend/src/layer/logic/api_command/command_handler.rs:176]: I see...

File: backend/src/layer/logic/api_command/image_preprocessing/mod.rs [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+            reader.set_format(format);
+        } else {
+            reader = reader
+                .with_guessed_format()
+                .map_err(ImagePreprocessingError::FormatGuessError)?;
+        }
+
+        // read image
+        let image = reader.decode()?;
+
```
Comment (Whatsuup) [inline backend/src/layer/logic/api_command/image_preprocessing/mod.rs:72]: This resizing does not effect the image quality in terms of stretching?

File: backend/migrations/20231010154028_local_image.down.sql [hunk 5]
```diff
@@ -0,0 +1 @@
+-- Add down migration script here
```
Comment (Whatsuup) [inline backend/migrations/20231010154028_local_image.down.sql:1]: Missing?

**Review by Whatsuup — APPROVED** [review]

File: backend/migrations/20231010154028_local_image.down.sql (hunk 5 above)
Comment (worldofjoni) [reply backend/migrations/20231010154028_local_image.down.sql:1]: nothing usefull to do here... we can not just invent image urls and flickr ids

File: backend/src/layer/logic/api_command/image_preprocessing/mod.rs (hunk 4 above)
Comment (worldofjoni) [reply backend/src/layer/logic/api_command/image_preprocessing/mod.rs:72]: the aspect ration is preserved it that is what you mean

**Pull request merged**